code/script.js

```diff
@@ -231,95 +260,201 @@ const setSecret = () => {
+  const category = questions.options[questions.selectedIndex].parentNode.label //variable for the category of the dropdown option group
 
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const value = questions.value //variable for the option value
 
   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
 }
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  selectQuestion() //To avoid the error of this function not being called if user selects the default from questions dropdown
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+ const {category, value} = currentQuestion
   if (category === 'hair' || category === 'eyes') {
+    if (value === secret.hair || value === secret.eyes) {
+      filterCharacters(true)
+    } else {
+      filterCharacters(false)
+    }
 
   } else if (category === 'accessories' || category === 'other') {
+    if (secret.accessories.includes(value) || (secret.other.includes(value))) {
+      filterCharacters(true)
+    } else {
+      filterCharacters(false)
+    }
 
   }
 }
 
-// It'll filter the characters array and redraw the game board.
+// Filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
+  countQuestionsAsked++
   const { category, value } = currentQuestion
+ 
   // Show the correct alert message for different categories
-  if (category === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
-    }
-  } else if (category === 'other') {
-    // Similar to the one above
+  if (countQuestionsAsked > questionsAllowed) {
+    alertify.alert(gameName, "You've run out of questions to ask. </br> Time to make an accusation!");
+  
   } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+    questionsAskedDisplay.innerText = countQuestionsAsked
+
+    if (category === 'hair') {
+      if (keep) {
+        alertify.alert(gameName,
+          `Yes, the suspect has ${value} hair! Keep all suspects that have ${value} hair.`
+        )
+      } else {
+        alertify.alert(gameName,
+          `No, the suspect doesn't have ${value} hair! Remove all suspects that have ${value} hair.`
```

Comment:
You could merge the hair and eyes conditions by  writing first:
`if (category === 'hair' || category === 'eyes')` and then changing the alert message like this:
``No, the suspect doesn't have ${value} hair! Remove all suspects that have ${value} ${category}.``